test/test_nn.py

```diff
@@ -2164,6 +2164,83 @@ def test_invalid_dropout_p(self):
         self.assertRaises(ValueError, lambda: F.dropout(v, -0.1))
         self.assertRaises(ValueError, lambda: F.dropout(v, 1.1))
 
+    def test_pad_sequence(self):
+        def pad(tensor, length):
+            return torch.cat(
+                [tensor.data, tensor.data.new(
+                    length - tensor.size(0), *tensor.size()[1:]).zero_()])
+        # single dimensional
+        a = Variable(torch.Tensor([1, 2, 3]))
+        b = Variable(torch.Tensor([4, 5]))
+        c = Variable(torch.Tensor([6]))
+
+        # batch_first = true
+        expected = Variable(torch.Tensor([[1, 2, 3], [4, 5, 0], [6, 0, 0]]))
+        padded = rnn_utils.pad_sequence([a, b, c], [3, 2, 1], True)
+        self.assertEqual(padded, expected)
+
+        # batch_first = false
+        padded = rnn_utils.pad_sequence([a, b, c], [3, 2, 1])
+        self.assertEqual(padded, expected.transpose(0, 1))
+
+        # more dimensional
+        sequences = []
+        lengths = []
+        maxlen = 9
+        for i in reversed(range(maxlen + 1)[1:]):
+            seq_len = i * i
+            lengths.append(seq_len)
+            sequences.append(Variable(torch.rand(seq_len, 400)))
+        expected = []
+        for seq in sequences:
+            expected.append(pad(seq, maxlen * maxlen))
+        # batch first = true
+        expected = Variable(torch.stack(expected))
+        padded = rnn_utils.pad_sequence(sequences, lengths, True)
+        self.assertEqual(padded, expected)
+
+        # batch first = false
+        padded = rnn_utils.pad_sequence(sequences, lengths)
+        self.assertEqual(padded, expected.transpose(0, 1))
+
+        # unsorted sequences should raise exception
+        self.assertRaises(
+            ValueError, lambda: rnn_utils.pad_sequence([b, a, c], [2, 3, 1]))
+
+    def test_pack_sequence(self):
+        # single dimensional
+        a = Variable(torch.Tensor([1, 2, 3]))
+        b = Variable(torch.Tensor([4, 5]))
+        c = Variable(torch.Tensor([6]))
+        packed = rnn_utils.pack_sequence([a, b, c], [3, 2, 1])
+        expected = torch.Tensor([1, 4, 6, 2, 5, 3])
+        self.assertEqual(packed.batch_sizes, [3, 2, 1])
+        self.assertEqual(packed.data.data, expected)
+
+        # more dimensions
+        ####################################
```

Comment:
Can you make this test more than one dimension? Just do a for loop for 1, 2, 3 trailing dims.